doc/spec/manifest.md

```diff
@@ -0,0 +1,245 @@
+# Distribution Content Manifests
+
+A Content Manifest is a simple JSON file which contains general fields that
+are typical to any package management system. The goal is for these manifests
+to describe an application and its dependencies in a content-addressable and
+verifiable way.
+
+## Field Descriptions
+
+- **`schemaVersion`** *int*
+
+    Specifies the schema version of the manifest as an integer. This document 
+    describes version `2` only.
+
+- **`createdAt`** *string*
+
+    ISO-8601 formatted combined date and time at which the manifest was
+    created.
+
+- **`maintainers`** *array*
+
+    An array of objects representing persons or organizations which are
+    designated as mainainers of the described content. Fields are:
+    
+    - **`name`** *string*
+    
+        The name of a maintaining person or organization.
+    
+    - **`email`** *string*
+    
+        The email address of a maintainer.
+    
+    - **`url`** *string*
+    
+        A URL to a maintainer's website, blog, etc.
+
+- **`description`** *string*
+
+    A brief description of the content or application.
+
+- **`keywords`** *array*
+
+    An array of strings which should be considered keywords that may be used
+    by a separate search system to index content.
+
+- **`website`** *string*
+
+    A URL to a related website or project page for the application.
+
+- **`licenses`** *array*
+
+    An array of strings which identify any software licenses which apply
+    to the application. Values should be a [SPDX License Identifier](https://spdx.org/licenses/).
+
+- **`Dependencies`** *array*
+
+    An array of objects which describe content which this application depends
+    on. Those may be generic blobs of data or other content manifests and
+    their dependencies as described by their media type. Fields are:
+    
+    - **`mediaType`** *string*
+    
+        The MIME type of the dependency object.
+    
+    - **`size`** *int*
+    
+        The size in bytes of the dependency object.
+    
+    - **`repository`** *string*
+    
+        The name of the repository from which to fetch the object.
+
+    - **`encryption`** *object*
+
+        An object containing information about the algorithm, key, and
+        initialization vector used to decrypt the content. If omitted,
+        the content should be assumed to be unencrypted. Fields are:
+
+        - **`algorithm`** *string*
+
+            Identifies the  encryption algorithm used.
+
+        - **`key`** *string*
```

Comment:
Yes, the idea is that you want to encrypt your dependency content, but because the keys will be in the manifest you must also encrypt the manifest. Then you only need to know the key to decrypt the manifest to get access to all of the separately-encrypted dependencies.

While this has some usability drawbacks (e.g., the hosting service not being able to intelligently index your content and dependencies) I think it's definitely a tradeoff some users would be willing to make.